keras_cv/models/backbones/vit/vit_backbone_presets.py

```diff
@@ -0,0 +1,289 @@
+        },
+    },
+    "vittiny32": {
+        "metadata": {
+            "description": (
+                "Vision Transformer Model with patch size of 32x32,"
+                "3 heads, and a hidden size of 192."
+            ),
+        },
+        "class_name": "keras_cv.models>ViTBackbone",
+        "config": {
+            "patch_size": 32,
+            "transformer_layer_num": 12,
+            "project_dim": 192,
+            "mlp_dim": 768,
+            "num_heads": 3,
+            "mlp_dropout": 0.0,
+            "attention_dropout": 0.0,
+        },
+    },
+    "vits32": {
+        "metadata": {
+            "description": (
+                "Vision Transformer Model with patch size of 32x32,"
+                "6 heads, and a hidden size of 384."
+            ),
+        },
+        "class_name": "keras_cv.models>ViTBackbone",
+        "config": {
+            "patch_size": 32,
+            "transformer_layer_num": 12,
+            "project_dim": 384,
+            "mlp_dim": 1536,
+            "num_heads": 6,
+            "mlp_dropout": 0.0,
+            "attention_dropout": 0.0,
+        },
+    },
+    "vitb32": {
+        "metadata": {
+            "description": (
+                "Vision Transformer Model with patch size of 32x32,"
+                "12 heads, and a hidden size of 768."
+            ),
+        },
+        "class_name": "keras_cv.models>ViTBackbone",
+        "config": {
+            "patch_size": 32,
+            "transformer_layer_num": 12,
+            "project_dim": 768,
+            "mlp_dim": 3072,
+            "num_heads": 12,
+            "mlp_dropout": 0.0,
+            "attention_dropout": 0.0,
+        },
+    },
+    "vitl32": {
+        "metadata": {
+            "description": (
+                "Vision Transformer Model with patch size of 32x32,"
+                "16 heads, and a hidden size of 1024."
+            ),
+        },
+        "class_name": "keras_cv.models>ViTBackbone",
+        "config": {
+            "patch_size": 32,
+            "transformer_layer_num": 24,
+            "project_dim": 1024,
+            "mlp_dim": 4096,
+            "num_heads": 16,
+            "mlp_dropout": 0.1,
+            "attention_dropout": 0.0,
+        },
+    },
+    "vith32": {
+        "metadata": {
+            "description": (
+                "Vision Transformer Model with patch size of 32x32,"
+                "16 heads, and a hidden size of 1280."
+            ),
+        },
+        "class_name": "keras_cv.models>ViTBackbone",
+        "config": {
+            "patch_size": 32,
+            "transformer_layer_num": 32,
+            "project_dim": 1280,
+            "mlp_dim": 5120,
+            "num_heads": 16,
+            "mlp_dropout": 0.1,
+            "attention_dropout": 0.0,
+        },
+    },
+}
+
+backbone_presets_with_weights = {
+    "vittiny16_imagenet": {
+        "metadata": {
+            "description": (
+                "Vision Transformer Model with patch size of 16x16,"
```

Comment:
If we know the top-1 accuracy of these models from their original source, we should include that here.

Reply:
It seems we do have that information. The weights were ported by @DavidLandup0 from the official [**google-research/vision_transformer**](https://github.com/google-research/vision_transformer/#available-vit-models) repo. I'll add that info here, thanks for the suggestion!

Reply:
Official [Weights link here](https://console.cloud.google.com/storage/browser/vit_models/augreg).

Reply:
Added for the values we do have the weights for!

Reply:
Yep - weights are ported from the original repo in JAX. The results matched between ImageNetV2's eval and a subsequent paper that included that metric. I don't have access to the original IN test set so that wasn't run before, but as V2 matches, I expect it to hold on the regular test set as well @suvadityamuk @ianstenbit